### PR TITLE
feat(api-reference): extend font weights for links and sidebar items

### PR DESCRIPTION
**Pull request by cameronrohani**
add font weights sidebar items (active, inactive) in a css variable
add link hover colours in a variable
add link weights in a variable
add link text decoration colours in a variable

### CODE REVIEW

File: packages/api-reference/src/components/Sidebar/Sidebar.vue [hunk 1]
```diff
@@ -183,6 +183,8 @@ onMounted(() => {
 <style scoped>
 .sidebar {
   --scalar-sidebar-indent-base: 12px;
+  --scalar-sidebar-font-weight-active: var(--scalar-semibold);
```
Comment (hwkr) [inline packages/api-reference/src/components/Sidebar/Sidebar.vue:186]: We don't want to define these in `variables.css`? We already have a few other sidebar variables in there 

https://github.com/scalar/scalar/blob/main/packages/themes/src/variables.css#L87-L89

**Review by hwkr — APPROVED** [review]
Looks good!

**Review by hanspagel — APPROVED** [review]
LGTM, but I agree with @hwkr’s comment

**Comment by hanspagel** [comment]
Cool, I think @hwkr’s comment was adressed. Let’s get this in!

**Pull request merged**